core/src/main/java/org/apache/james/core/healthcheck/Result.java

```diff
@@ -26,25 +26,66 @@
 public class Result {
 
     public static Result healthy(ComponentName componentName) {
-        return new Result(componentName, ResultStatus.HEALTHY, Optional.empty());
+        return Builder.builder().componentName(componentName).status(ResultStatus.HEALTHY).build();
```

Comment:
why did you wrote a builder if you don't let callers use it?

Reply:
I wanted you guys to validate the idea behind this approach first, so i didn't really have time to double check carefully. Sorry.